paperqa/agents/__init__.py

```diff
@@ -0,0 +1,550 @@
+from __future__ import annotations
+
+import ast
+import logging
+import operator
+import os
+import shutil
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import yaml
+from typing_extensions import Annotated
+
+from .. import __version__
+from ..utils import get_loop, pqa_directory
+
+try:
+    import anyio
+    import typer
+    from rich.console import Console
+    from rich.logging import RichHandler
+
+    from .main import agent_query, search
+    from .models import AnswerResponse, MismatchedModelsError, QueryRequest
+    from .search import SearchIndex, get_directory_index
+
+except ImportError as e:
+    raise ImportError(
+        '"agents" module is not installed please install it using "pip install paper-qa[agents]"'
+    ) from e
+
+app = typer.Typer()
+
+
+def configure_agent_logging(
+    verbosity: int = 0, default_level: int = logging.INFO
+) -> None:
+    """Default to INFO level, but suppress loquacious loggers."""
+    verbosity_map = {
+        0: {
+            "paperqa.agents.docs": logging.WARNING,
+            "paperqa.agents.main": logging.WARNING,
+            "anthropic": logging.WARNING,
+            "openai": logging.WARNING,
+            "httpx": logging.WARNING,
+            "paperqa.agents.models": logging.WARNING,
+        }
+    }
+
+    verbosity_map[1] = verbosity_map[0] | {
+        "paperqa.agents.main": logging.INFO,
+        "paperqa.models": logging.INFO,
+    }
+
+    verbosity_map[2] = verbosity_map[1] | {
+        "paperqa.agents.docs": logging.DEBUG,
+        "paperqa.agents.main": logging.DEBUG,
+        "paperqa.agents.main.agent_callers": logging.DEBUG,
+        "paperqa.models": logging.DEBUG,
+        "paperqa.agents.search": logging.DEBUG,
+    }
+
+    rich_handler = RichHandler(
+        rich_tracebacks=True,
+        markup=True,
+        show_path=False,
+        show_level=False,
+        console=Console(force_terminal=True),
+    )
+
+    rich_handler.setFormatter(logging.Formatter("%(message)s", datefmt="[%X]"))
+
+    root_logger = logging.getLogger()
```

Comment:
Why are we manipulating the root logger? Can we just use the `paperqa` loggers (and its descendents)? So that would be `logging.getLogger("paperqa")`